core/tests/integration/inputs/typecheck/dictionary_subtyping.ncl

```diff
@@ -0,0 +1,6 @@
+# test.type = 'pass'
+let test : {_ : {foo : Number}} = {a = {foo = 5}} in
+let test_func : {_ : {_ : Number}} -> {_ : {_ : Number}} = fun a => a in 
+let result : {_ : {_ : Number}} = test_func test in 
+true
```

Comment:
Once again, I think we don't have to write all annotations explicitly and rely more on type inference.

```suggestion
let test : _ = 
  let test_func : {_ : {_ : Number}} -> {_ : {_ : Number}} = fun a => a in 
  test_func {a = {foo = 5}}
in
true
```